src/cmd/disas/main.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2015 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+
+	"cmd/internal/goobj"
+	"cmd/internal/obj"
+)
+
+func objdumpcmd(fname string) *exec.Cmd {
```

Comment:
I think this was mentioned below. Perhaps this could return an error:
func objdumpcmd(fname string) (*exec.Cmd, error) {
switch GOARCH {
	case "arm":
		...
	case "riscv":
		...
	default:
		return nil, fmt.Errorf(""unsupported architecture %s", GOARCH)
	}
}

And it could be used below:
// Ensure that we actually support this architecture.
if _, err := objdumpcmd(""); err != nil {
	log.Fatalf("%s\n", err)
}

Or something along those lines. Alternatively you could factor this out into a map of arch string->cmd and test for existence in the map to determine if an arch is supported instead of calling the function with a dummy parameter.